app/src/main/java/com/portfolio/tasky/views/TaskyAppCompatEditText.kt

```diff
@@ -1,62 +1,117 @@
 package com.portfolio.tasky.views
 
+import android.R.attr.*
 import android.annotation.SuppressLint
 import android.content.Context
 import android.content.res.TypedArray
 import android.graphics.drawable.Drawable
 import android.text.InputType
+import android.text.Selection
 import android.util.AttributeSet
 import android.view.LayoutInflater
+import android.view.MotionEvent
+import android.view.View
+import android.view.View.OnTouchListener
 import android.widget.FrameLayout
 import androidx.appcompat.widget.AppCompatEditText
+import androidx.core.content.ContextCompat
 import com.portfolio.tasky.R
 import com.portfolio.tasky.databinding.LayoutInputFieldBinding
 
-class TaskyAppCompatEditText(context: Context, attrs: AttributeSet) : FrameLayout(context, attrs)   {
 
-    private lateinit var viewBinding: LayoutInputFieldBinding
+class TaskyAppCompatEditText(context: Context, attrs: AttributeSet) : FrameLayout(context, attrs), OnTouchListener   {
+
+    lateinit var subLayout: LayoutInputFieldBinding
     private var error : Boolean = false
+    private var isPassword : Boolean = false
     private var drawableLast : Drawable? = null
     private lateinit var editText : AppCompatEditText
     private var inputType : Int = 0
     private var hint : String? = null
 
+
     init {
         setupView(attrs)
-        isError()
+        setError(false)
         setDrawableLast()
         setInputType()
         setHint()
+        setValid(false)
+        isPassword()
```

Comment:
This is just an idea
Maybe make a different EditText, that inherits from this one, but make it only for password

Something like

TaskyAppCompatPasswordEditText

So you have 2 edit texts with different logic

To be 100% honest, I haven't done many custom views on XML (I did couple years ago) so I might be a bit rusty with it
But should work

Reply:
We can use another AppCompatEdittext for Passwords but that would just require to handle few more files. In the current implementation, the custom field handles both inputTypes, TEXT & PASSWORD.

Reply:
Sounds good. Let's leave it like it is then